src/driver-hidpp20.c

```diff
@@ -1179,6 +1179,34 @@ hidpp20drv_read_profile_8100(struct ratbag_profile *profile)
 	ratbag_profile_set_report_rate(profile, p->report_rate);
 }
 
+static int
+hidpp20drv_init_profile_8100(struct ratbag_device *device)
+{
+	struct hidpp20drv_data *drv_data = ratbag_get_drv_data(device);
+	struct ratbag *ratbag = device->ratbag;
+	int rc;
+
+	log_debug(ratbag, "initializing onboard profiles\n");
+	rc = hidpp20_onboard_profiles_allocate(drv_data->dev, &drv_data->profiles);
+	if (rc < 0)
+		return rc;
+
+	rc = hidpp20_onboard_profiles_initialize(drv_data->dev, drv_data->profiles);
+	if (rc < 0)
+		return rc;
+
+	drv_data->num_profiles = drv_data->profiles->num_profiles;
+	drv_data->num_buttons = drv_data->profiles->num_buttons;
+
+	if (drv_data->capabilities & HIDPP_CAP_SWITCHABLE_RESOLUTION_2201)
+		drv_data->num_resolutions = drv_data->profiles->num_modes;
```

Comment:
To be clearer, I added this check to the original code. This check needs the capabilities to be populated first. The original code was being run in the loop where we parse the features and populate the capabilities. It would work if the `AdjustableDpi` feature was found before `OnboardMemoryProfiles`.